llama_index/retrievers/image_retriever.py

```diff
@@ -0,0 +1,54 @@
+from abc import abstractmethod
+from typing import List
+
+from llama_index.indices.query.schema import QueryBundle, QueryType
+from llama_index.prompts.mixin import PromptMixin
+from llama_index.schema import NodeWithScore
+
+
+class BaseImageRetriever(PromptMixin):
+    """Base Image Retriever Abstraction."""
+
+    def image_retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
```

Comment:
I remember this handling both retrieving with text or retrieving with an image -- but not anymore?

Reply:
discussed offline. 
if user want to do image to image retrieval, they need to set `image_path` in `query_bundle`
and call `embedding_image` to embed image